aten/src/ATen/native/mps/operations/UnaryOps.mm

```diff
@@ -81,10 +81,10 @@ void unary_op(const Tensor& self,
       newCachedGraph->outputTensor_ = unaryBlock(mpsGraph, castTensor);
     });
 
-    // If self is densely mapped in storage, create a dense output-like representation
+    // If self is non-densely mapped in storage, create a dense representation
     at::Tensor self_;
     if (!is_dense_in_storage(self)) {
-      self_ = at::empty_like(output);
+      self_ = at::empty_like(self);
```

Comment:
This is wrong (and I hope tests will capture this): it needs to re-stride self to be output like, but preserve dtype (and this is where the problem is)

Reply:
great catch! Yea I knew the issue was the dtype but I didn't think about the striding being copied over, I learned something about tensors today